server/game/cards/03-WC/Infomancer.js

```diff
@@ -0,0 +1,31 @@
+const Card = require('../../Card.js');
+
+class Infomancer extends Card {
+    setupCardAbilities(ability) {
+        this.play({
+            target: {
+                location: 'hand',
+                controller: 'self',
+                cardType: 'action',
+                gameAction: ability.actions.graft(context => ({ parent: context.source }))
+            }
+        });
+
+        this.reap({
+            effect: 'trigger the play effect of {0}',
+            target: {
+                location: 'any',
+                controller: 'self',
+                cardType: 'action',
+                cardCondition: (card, context) => card.parent === context.source && card.facedown === false,
```

Comment:
Why the facedown check?  Is this to target grafted cards?  Surely there's a more explicit way to do that?

Reply:
I think it's because there's a possibility of using Creed of Nurture to give Jargogle Infomancer's abilities.  We use the same system for both Jargogle's Play/Destroyed effects, and Infomancer's abilities, so we need to avoid the game treating a facedown card as a grafted one

Reply:
I get that, but I think it should be more explicit like some kind of grafted state or flag. That way the next time they had a place under effect we won’t have to rework everything

Reply:
Using `grafted` location on the card code now